library/src/main/java/com/wultra/android/mtokensdk/api/operation/AttributeTypeAdapter.kt

```diff
@@ -94,18 +94,22 @@ internal class AttributeTypeAdapter : TypeAdapter<Attribute>() {
         fun build(): Attribute? {
             return when (type) {
                 Attribute.Type.AMOUNT -> {
-                    val amount: BigDecimal = attr("amount") ?: return null
-                    val currency: String = attr("currency") ?: return null
-                    AmountAttribute(amount, currency, attr("amountFormatted"), attr("currencyFormatted"), attr("valueFormatted"), label)
+                    // For backward compatibility with legacy implementation, where the `amountFormatted` and `currencyFormatted` values might not be present,
+                    // we decode from `amount` and `currency` which were not nullable
+                    val amountFormatted: String = attr("amountFormatted") ?: attr<String>("amount").toString()
```

Comment:
`attr<String>("amount").toString()` Doesn't make sense - "get string and make toString" is redundant.

Also, we should keep the old behavior when null otherwise, this will crash.

```
val amountFormatted: String = attr("amountFormatted") ?: attr<BigDecimal>("amount")?.toString() ?: return null
```